packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -67,17 +74,21 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
   @override
   Widget build(BuildContext context) {
     final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
     context.watch<LocalDatabase>();
     final ThemeData themeData = Theme.of(context);
     _product = _localDatabase.upToDate.getLocalUpToDate(_initialProduct);
     final bool empty = _uploadedImages.isEmpty && _otherUploadedImages.isEmpty;
+
+    _addToHistory();
+
     return SmoothScaffold(
       appBar: AppBar(
         title: Text(appLocalizations.new_product),
         automaticallyImplyLeading: empty,
       ),
       floatingActionButton: FloatingActionButton.extended(
-        onPressed: () async => Navigator.maybePop(context),
+        onPressed: () => Navigator.maybePop(context),
```

Comment:
```suggestion
        onPressed: () async => Navigator.maybePop(context),
```

`Navigator.maybePop(context)` is a `Future`.